WebApp/public/scripts/register-events.js

```diff
@@ -199,51 +205,81 @@ export function registerGamepadEvents(videoPlayer) {
     data.setFloat64(11, e.y, true);
     _videoPlayer && _videoPlayer.sendMsg(data.buffer);
   }
+
+  gamepadConnected = (e) => { gamepadHandler(e, true); }
+  gamepadDisconnected = (e) => { gamepadHandler(e, false); }
+
+
+  document.addEventListener("gamepadButtonDown", sendGamepadButtonDown, false);
+  document.addEventListener("gamepadButtonUp", sendGamepadButtonUp, false);
+  document.addEventListener("gamepadButtonPressed", sendGamepadButtonPressed, false);
+  document.addEventListener("gamepadAxis", gamepadAxisChange, false);
+
+  window.addEventListener("gamepadconnected", gamepadConnected, false);
+  window.addEventListener("gamepaddisconnected", gamepadDisconnected, false);
 }
 
+export function unregisterGamepadEvents(videoPlayer) {
+
+    document.removeEventListener("gamepadButtonDown", sendGamepadButtonDown, false);
+    document.removeEventListener("gamepadButtonUp", sendGamepadButtonUp, false);
+    document.removeEventListener("gamepadButtonPressed", sendGamepadButtonPressed, false);
+    document.removeEventListener("gamepadAxis", gamepadAxisChange, false);
+
+    window.removeEventListener("gamepadconnected", gamepadConnected, false);
+    window.removeEventListener("gamepaddisconnected", gamepadDisconnected, false);
+
+}
+
+
+let sendKeyUp = undefined;
+let sendKeyDown = undefined;
+
 
 export function registerKeyboardEvents(videoPlayer) {
+
   const _videoPlayer = videoPlayer;
-  document.addEventListener('keyup', sendKeyUp, false);
-  document.addEventListener('keydown', sendKeyDown, false);
 
-  function sendKeyUp(e) {
+  function sendKey(e, type) {
+    const key = Keymap[e.code];
+    const character = e.key.length === 1 ? e.key.charCodeAt(0) : 0;
+    //console.log("key down " + key + ", repeat = " + e.repeat + ", character = " + character);
```

Comment:
I think choose delete comment or leave log.
( also other comment out about `console.log`.)

Reply:
Oh I think it can be handy to keep them commented as an available tool for debugging purposes, but we can clean them completely if you prefer.

Reply:
I see, I support keeping it for Debug too.
thanks answer